### DISCUSSION TITLE
Is is possible to setup a mock that confirms that no methods are called?

### DISCUSSION THREAD

**Discussion by wt** [asomers/mockall#559] [Questions]
I have an object that I want to make sure it doesn't call methods on the mock in some cases. Basically, I have a trait for different backing stores for a small bit of data. I want to make sure that the data is lazily populated, so I want to make sure that certain actions have no calls to the mocked trait.

My current try looks like this in the test code:

```
let mut mock = MockTrait::new();
// make sure all methods are listed
// make sure to update on added methods in the trait
mock
    .expect_create()
    .never();
mock
    .expect_read()
    .never();
mock
    .expect_update()
    .never();
mock
    .expect_delete()
    .never();
// fails in this case, but I'd like to not have to manually list all the methods above
mock.create(); 
```

I guess I could manually impl a method on the mock type that sets this up the expectations and use that, but I was wondering if there was something built-in that I could use.

I have looked around the api, and it looks like there is not a built-in way to do that. I just wanted to ask here to see if I missed something.

**Comment by asomers** [comment]
No, you're doing it the best way.  Technically, you don't even need to do anything, because a method with no expectations will always panic when called.  But doing it this way and using `never` will produce better error messages.

**Reply by wt** [reply]
The default error message is quite awful. Would it make more sense to have the default be an `.never()`-like expectation on every method so that the better error message is the default?

**Reply by asomers** [reply]
Not really, not when you consider that each Expectation also includes argument matchers.  So it is possible to assert for example that an expectation be called 1 time with one set of arguments, and never with a different set.  The current default error message "Expectation not found" is the most general.

**Reply by wt** [reply]
I see. Thanks for the help.
